src/consumer/context.rs

```diff
@@ -1,26 +1,24 @@
 use std::{
     any::{Any, TypeId},
     collections::HashMap,
+    sync::Arc,
 };
 
 use crate::{
     codec::Codec,
-    message::{
-        Message, RawHeaders, RawMessage, TryFromRawHeaders, CONTENT_TYPE_HEADER, KIND_HEADER,
-    },
+    message::{RawHeaders, RawMessage, KIND_HEADER},
 };
 
 use super::{Extensions, MessageBus};
+use crate::consumer::message_bus::IncomingMessage;
 
 use thiserror::Error;
 
-pub struct ProcessContext<'a, B: MessageBus> {
-    payload: &'a [u8],
-    key: Option<&'a [u8]>,
-    headers: RawHeaders,
+pub struct ProcessContext<'a, B: MessageBus, C: Codec> {
+    message: Arc<B::IncomingMessage>,
     extensions: &'a Extensions,
     cache: &'a mut LocalCache,
-    raw_msg: &'a B::IncomingMessage,
+    pub codec: &'a C,
```

Comment:
Other members have corresponding getter methods, I think we should stick to this style.